apps/obsidian/src/utils/createNode.ts

```diff
@@ -130,9 +132,29 @@ export const processTextToDiscourseNode = async ({
     formattedNodeName,
     nodeType,
   });
-  if (newFile) {
+
+  if (newFile && editor) {
     editor.replaceSelection(`[[${formattedNodeName}]]`);
   }
 
   return newFile;
 };
+
+export const processTextToDiscourseNode = async ({
```

Comment:
We don't need an extra function for this.  We could just use `createDiscourseNode`, make `text` optional and grab the selection if it's null.  `editor` can be required.

```typescript
const nodeText = text ?? editor.getSelection().trim();
```